exchanges/btcmarkets/btcmarkets_test.go

```diff
@@ -20,487 +19,413 @@ const (
 	canManipulateRealOrders = false
 )
 
-func TestSetDefaults(t *testing.T) {
+func TestMain(m *testing.M) {
 	b.SetDefaults()
-}
-
-func TestSetup(t *testing.T) {
 	cfg := config.GetConfig()
 	err := cfg.LoadConfig("../../testdata/configtest.json", true)
 	if err != nil {
-		t.Fatal("BTC Markets load config error", err)
+		log.Fatal("BTC Markets load config error", err)
 	}
 	bConfig, err := cfg.GetExchangeConfig("BTC Markets")
 	if err != nil {
-		t.Error("BTC Markets Setup() init error")
+		log.Fatalf("BTC Markets Setup() init error: %v", err)
 	}
 	bConfig.API.Credentials.Key = apiKey
 	bConfig.API.Credentials.Secret = apiSecret
 	bConfig.API.AuthenticatedSupport = true
 
 	err = b.Setup(bConfig)
 	if err != nil {
-		t.Fatal("BTC Markets setup error", err)
+		log.Fatal("BTC Markets setup error", err)
 	}
+
+	os.Exit(m.Run())
 }
 
-func TestGetMarkets(t *testing.T) {
-	t.Parallel()
-	_, err := b.GetMarkets()
-	if err != nil {
-		t.Error("GetMarkets() error", err)
+func areTestAPIKeysSet() bool {
+	if b.API.Credentials.Key != "" && b.API.Credentials.Key != "Key" &&
+		b.API.Credentials.Secret != "" && b.API.Credentials.Secret != "Secret" {
+		return true
 	}
+	return false
 }
 
-func TestGetTicker(t *testing.T) {
+func TestGetMarkets(t *testing.T) {
```

Comment:
Across this testing suite can you please add in test cases that cause the endpoints to fail and you are expecting and error.